pkg/agent/openflow/pipeline.go

```diff
@@ -120,17 +132,30 @@ func (rt regType) reg() string {
 const (
 	// marksReg stores traffic-source mark and pod-found mark.
 	// traffic-source resides in [0..15], pod-found resides in [16].
-	marksReg     regType = 0
-	portCacheReg regType = 1
-	swapReg      regType = 2
+	marksReg        regType = 0
+	portCacheReg    regType = 1
+	swapReg         regType = 2
+	endpointIPReg   regType = 3               // Use reg3 to store endpoint IP
+	endpointPortReg regType = 4               // Use reg4[0..15] to store endpoint port
+	serviceLearnReg         = endpointPortReg // Use reg4[16..18] to store endpoint selection states.
+	// marksRegServiceNeedLB indicates a packet need to do service selection.
+	marksRegServiceNeedLB uint32 = 0b001
+	// marksRegServiceHitted indicates a packet has done service selection.
+	marksRegServiceHitted uint32 = 0b010
```

Comment:
"Hitted" is a pretty non-standard word ("hit" is the common form). Maybe "Selected" would make more sense based on your comments?

Reply:
Thanks, updated.